Tasks/NuGetCommandV2/nugetpublisher.ts

```diff
@@ -466,4 +483,37 @@ function getAccessToken(isInternalFeed: boolean): string{
     }
 
     return accessToken;
-}
\ No newline at end of file
+}
+
+async function tryServiceConnection(endpoint: EndpointCredentials, feed: any) : Promise<boolean>
+{
+    // Create request
+    const request = new WebRequest();
+    const token64 = Buffer.from(`${endpoint.username}:${endpoint.password}`).toString('base64');
+    request.uri = endpoint.endpoint;
+    request.method = 'GET';
+    request.headers = {
+        "Content-Type": "application/json",
+        "Authorization": "Basic " + token64
+    };
+
+    return await sendRequest(request).then((response: WebResponse) => {
```

Comment:
Don't mix async/await and explicit promises:

```suggestion
const response = await sendRequest(request);
```

(you'll also need to fix up the end of the method)

Reply:
Whats the reasoning behind this?

Reply:
TBH really I should make the stronger statement "Don't use explicit promises unless you have a very specific reason to. Use async/await instead." It's simpler to read, easier to reason about, harder to get wrong, and "essentially" translated by the interpreter into promises anyway. Mixing the two like this isn't semantically incorrect, but it smells bad.